apps/meteor/app/livechat/server/lib/QueueManager.ts

```diff
@@ -255,7 +257,17 @@ export class QueueManager {
 			void notifyOnSettingChanged(livechatSetting);
 		}
 
-		const newRoom = (await this.queueInquiry(inquiry, room, defaultAgent)) ?? (await LivechatRooms.findOneById(rid));
+		let newRoom;
+		// Note: if the contact is not verified, we cannot add it to the queue, since no agent should be able to pick the conversation
+		//       up. So if the contact is not verified we don't add it to queue and wait an event to be sent to properly add it to queue
+		if (inquiry.v.contactId && (await shouldTriggerVerificationApp(inquiry.v.contactId, room.source))) {
+			newRoom = await LivechatRooms.findOneById(rid);
+			await LivechatContacts.updateContactChannel(inquiry.v._id, { verified: true });
+			// void Apps.self?.triggerEvent(AppEvents.IPostContactVerificationAppAssigned, inquiry.v, room);
```

Comment:
```suggestion
```
Also, comments mentions something about an event, but there's no event listening in here 👀

Reply:
@Gustrb decided to remove this line and the comment too since it's not quite right